src/pr-trigger.ts

```diff
@@ -5,158 +5,146 @@ import { process as computeActions } from "./compute-pr-actions";
 import { executePrActions } from "./execute-pr-actions";
 import { mergeCodeOwnersOnGreen } from "./side-effects/merge-codeowner-prs";
 import { HttpRequest, Context } from "@azure/functions";
-import { Webhooks, EventPayloads } from "@octokit/webhooks";
+import { createEventHandler, EmitterWebhookEvent, verify } from "@octokit/webhooks";
 
 const prHandlers: { [prNumber: number]: () => void } = {};
 
 export async function httpTrigger(context: Context, req: HttpRequest) {
-
-    if (!(process.env["BOT_AUTH_TOKEN"] || process.env["AUTH_TOKEN"])) {
-        throw new Error("Set either BOT_AUTH_TOKEN or AUTH_TOKEN to a valid auth token");
-    }
-
     context.log(`[${process.version}] HTTP trigger function received a request.`);
 
-    const event = req.headers["x-github-event"];
-    if (!event) throw new Error(`Did not get a x-github-event header`);
-
     const isDev = process.env.AZURE_FUNCTIONS_ENVIRONMENT === "Development";
     const secret = process.env.GITHUB_WEBHOOK_SECRET;
-    const webhooks = new Webhooks({ secret });
 
     // For process.env.GITHUB_WEBHOOK_SECRET see
     // https://ms.portal.azure.com/#blade/WebsitesExtension/FunctionsIFrameBlade/id/%2Fsubscriptions%2F57bfeeed-c34a-4ffd-a06b-ccff27ac91b8%2FresourceGroups%2Fdtmergebot%2Fproviders%2FMicrosoft.Web%2Fsites%2FDTMergeBot
-    if (!isDev && !webhooks.verify(req.body, req.headers["x-hub-signature-256"]!)) {
+    if (!isDev && !verify(secret!, req.body, req.headers["x-hub-signature-256"]!)) {
```

Comment:
Use the standalone method because we're not using either `webhooks.middleware()` or `webhooks.verifyAndReceive()` https://github.com/octokit/webhooks.js/tree/master/src/verify

Reply:
As far as I know that code only supports Node.js' `http.createServer()`, not an Azure Functions `HttpRequest`.

Reply:
OK